From: SeongJae Park <sj@kernel.org>
Subject: mm/damon/lru_sort: add inactive mem ratio quota goal to cold_scheme
Date: Tue, 13 Jan 2026 21:48:53 -0800

damon_lru_sort_add_quota_goals() is adding inactive memory ratio quota
goal, which is for cold pages scheme, to hot pages scheme.  Fix it.

Link: https://lkml.kernel.org/r/20260114055308.79884-1-sj@kernel.org
Signed-off-by: SeongJae Park <sj@kernel.org>
Reported-by: wang lian <lianux.mm@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/lru_sort.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/damon/lru_sort.c~mm-damon-lru_sort-support-active-inactive-memory-ratio-based-auto-tuning-fix
+++ a/mm/damon/lru_sort.c
@@ -238,7 +238,7 @@ static int damon_lru_sort_add_quota_goal
 			10000 - active_mem_bp + 2);
 	if (!goal)
 		return -ENOMEM;
-	damos_add_quota_goal(&hot_scheme->quota, goal);
+	damos_add_quota_goal(&cold_scheme->quota, goal);
 	return 0;
 }
 
_
